core/src/main/java/org/openstack4j/openstack/artifact/internal/BaseArtifactServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package org.openstack4j.openstack.artifact.internal;
+
+import org.openstack4j.api.types.ServiceType;
+import org.openstack4j.core.transport.HttpResponse;
+import org.openstack4j.model.artifact.*;
+import org.openstack4j.model.artifact.builder.ArtifactUpdateBuilder;
+import org.openstack4j.model.common.ActionResponse;
+import org.openstack4j.model.common.Payload;
+import org.openstack4j.model.common.payloads.FilePayload;
+import org.openstack4j.openstack.artifact.domain.ArtifactUpdateModel;
+import org.openstack4j.openstack.common.ListEntity;
+import org.openstack4j.openstack.common.functions.EnforceVersionToURL;
+import org.openstack4j.openstack.internal.BaseOpenStackService;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.openstack4j.core.transport.ClientConstants.CONTENT_TYPE_ARTIFACT_PATCH;
+import static org.openstack4j.core.transport.ClientConstants.CONTENT_TYPE_OCTECT_STREAM;
+import static org.openstack4j.core.transport.ClientConstants.HEADER_ACCEPT;
+
+/**
+ * Created by vadavi on 18-01-2017.
+ */
+public class BaseArtifactServiceImpl extends BaseOpenStackService {
+    private ArtifactType artifactType;
+
+    protected BaseArtifactServiceImpl(ArtifactType artifactType) {
+
+        super(ServiceType.ARTIFACT, EnforceVersionToURL.instance(""));
+        this.artifactType = artifactType;
+
+    }
+
+    protected <T> T list(Class<T> clazz) {
+
+        return get(clazz,uri("/artifacts/"+ artifactType.value())).execute();
```

Comment:
Could you add the path `/artifacts/` to https://github.com/ContainX/openstack4j/blob/master/core/src/main/java/org/openstack4j/core/transport/ClientConstants.java#L32 and reference that instead?
Then search&replace in other methods.